src/networktables/NetworkTable.cpp

```diff
@@ -43,6 +42,8 @@ void NetworkTable::SetTeam(int team) {
   char tmp[30];
 #ifdef _MSC_VER
   sprintf_s(tmp, "roboRIO-%d-FRC.local\n", team);
+#elif defined(__ANDROID__)
+  snprintf(tmp, 30, "roboRIO-%d-FRC.local\n",team);
```

Comment:
I used the directions here http://developer.android.com/ndk/guides/standalone_toolchain.html to create a standalone toolchain for android. I believe I used this switch ```--toolchain=arm-linux-androideabi-4.9``` to get the 4.9 version I was using. Then what I did was placed the standalone toolchain in my path, and then changed the arm.gradle file to use the different toolchain prefix. Then just compiled with gradle, and it worked (well kind of). I tested on my nexus 6 with Xamarin, and it worked and connected. However, MDNS did not work, and for some reason the code had be in debug mode, as release mode would not connect to the network properly. 

I was kind of surprised that it worked too. But the changes here are the only changes I needed to get it working and compiling.